corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -283,7 +251,33 @@ def _update_details_based_on_position(self, list_rows, short_details, detail_row
             self._update_detail(row, detail)
 
     def _partial_upload(self, rows, details):
+        expected_fields = [detail.field for detail in details]
+        received_fields = [row['id'] for row in rows]
+        message = _('The following rows are missing for menu {index}: {fields}').format(
+            index=self.module.id + 1,
+            fields=', '.join(set(expected_fields) - set(received_fields))
+        )
+        self.msgs.append((messages.info, message))
+
+        expected_field_counter = Counter(expected_fields)
+        received_field_counter = Counter(received_fields)
         for detail, row in zip_with_gaps(details, rows,
                                          lambda detail: detail.field,
                                          lambda row: row['id']):
+            field = row['id']
+            if (
+                received_field_counter[field] > 1 and
+                received_field_counter[field] != expected_field_counter[field]
+            ):
+                message = _(
+                    'There is more than one translation for case property '
+                    '"{field}" for menu {index}, but some translations are '
+                    'missing. Unable to determine which translation(s) to '
+                    'use. Skipping this case property.'
+                ).format(
+                    index=self.module.id + 1,
+                    field=row.get('case_property', '')
+                )
+                self.msgs.append((messages.info, message))
```

Comment:
Nitpick: I'd use `messages.error`, that's what the other messages that notify about skipped translations do.

Reply:
For ICDS this would succeed silently. I'm happy to use `messages.error` for the sake of consistency, but I added this just to let the user know what fields still need to be translated. We could also drop it to keep with current behaviour.

Reply:
Sorry. I was confusing this comment with your comment on line 260. I agree, this should be `messages.error`.